nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -609,23 +609,38 @@ sub multiLineList {
 ${\join "", (map { "  $_\n" } (uniq @attrs))}}
 EOF
 
-sub generateNetworkingDhcpConfig {
-    # FIXME disable networking.useDHCP by default when switching to networkd.
+sub generateNetworkingConfig {
     my $config = <<EOF;
-  # Enables DHCP on each ethernet and wireless interface. In case of scripted networking
-  # (the default) this is the recommended approach. When using systemd-networkd it's
-  # still possible to use this option, but it's recommended to use it in conjunction
-  # with explicit per-interface declarations with `networking.interfaces.<interface>.useDHCP`.
-  networking.useDHCP = lib.mkDefault true;
+  # Use systemd-networkd to enable DHCP (IPv4) and Router
+  # Advertisements (IPv6) on every ethernet and wireless interface.
+  systemd.network = {
+    enable = true;
+    networks = {
+      # Use lower numbers for additional profiles, since networkd
+      # processes these alphanumerically and every interface will only
+      # ever use its first match.
+      "99-autoconfig-ether" = {
+        matchConfig = {
+          Kind = "!*";
+          Type = [ "ether" ];
+        };
+        networkConfig = {
+          DHCP = "ipv4";
+          IPv6AcceptRA = true;
+        };
+        linkConfig.RequiredForOnline = "no";
+      };
+      "99-autoconfig-wlan" = {
+        matchConfig.Type = [ "wlan" ];
+        networkConfig = {
+          DHCP = "ipv4";
+          IPv6AcceptRA = true;
+        };
+        linkConfig.RequiredForOnline = "no";
```

Comment:
This should be removed I think?  Scripted networking actually respects `network-online.target`  as  `dhcpcd.service` is only ready after the first DHCP lease is received.  So we also want `network-online.target` to work keep working when we transition to networkd.

Reply:
( I fat-fingered the approve button. I just wanted to leave a comment).